pkg/management/postgres/instance.go

```diff
@@ -247,11 +251,41 @@ func (instance *Instance) SetCanCheckReadiness(enabled bool) {
 	instance.canCheckReadiness.Store(enabled)
 }
 
+// HasWALDiskSpace checks if we have enough disk space to store two WAL files
+func (instance *Instance) HasWALDiskSpace(ctx context.Context) (bool, error) {
+	pgControlDataString, err := instance.GetPgControldata()
+	if err != nil {
+		return false, fmt.Errorf("while running pg_controldata to detect WAL segment size: %w", err)
+	}
+
+	pgControlData := utils.ParsePgControldataOutput(pgControlDataString)
+	walSegmentSizeString, ok := pgControlData["Bytes per WAL segment"]
+	if !ok {
+		return false, fmt.Errorf("no 'Bytes per WAL segment' secton into pg_controldata output")
+	}
+
+	walSegmentSize, err := strconv.Atoi(walSegmentSizeString)
+	if err != nil {
+		return false, fmt.Errorf(
+			"wrong 'Bytes per WAL segment' pg_controldata value (not an integer): '%s' %w",
+			walSegmentSizeString, err)
+	}
+
+	walDirectory := path.Join(instance.PgData, pgWalDirectory)
+	return fileutils.NewDirectory(walDirectory).HasSpaceInDirectory(ctx, walSegmentSize*2)
+}
+
 // SetMightBeUnavailable marks whether the instance being down should be tolerated
 func (instance *Instance) SetMightBeUnavailable(enabled bool) {
 	instance.mightBeUnavailable.Store(enabled)
 }
 
+// SetNoDiskSpaceLeft marks whether PostgreSQL exited because there's no more
+// disk space left for WALs
+func (instance *Instance) SetNoDiskSpaceLeft(enabled bool) {
+	instance.noWALDiskSpaceLeft.Store(enabled)
+}
+
```

Comment:
Firstly, the method is hard to parse, since it begins with "set no disk ..." which gives a wrong impression.
Then, `enabled` doesn't look like the correct name for this boolean.

```suggestion
// SetDiskIsFull marks whether PostgreSQL exited because there's no more
// disk space left for WALs
func (instance *Instance) SetDiskIsFull(diskIsFull bool) {
	instance.noWALDiskSpaceLeft.Store(diskIsFull)
}

```

Reply:
I renamed it `SetWALDiskIsFull`, is that fine with you?

Reply:
Yes, perfect